pkg/ssh/keymanager_test.go

```diff
@@ -381,35 +388,100 @@ func TestKeyManager_EnsureKeysExist(t *testing.T) {
 	}
 }
 
-func mockPDC(t *testing.T, method, path string, code int) (u *url.URL, called *bool) {
-	t.Helper()
+func TestBackgroundRefresh(t *testing.T) {
+	t.Run("refresh is 0, do not refresh", func(t *testing.T) {
+		ctx := context.Background()
+		sut := testKeyManager(t)
+		sut.sshCfg.CertCheckCertExpiryPeriod = 0
 
-	called = new(bool)
-
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		assert.Equal(t, method, r.Method)
-		assert.Equal(t, path, r.URL.Path)
-		*called = true
-
-		resp := struct {
-			KnownHosts  string `json:"known_hosts"`
-			Certificate string `json:"certificate"`
-		}{
-			KnownHosts:  knownHosts,
-			Certificate: expectedCert,
-		}
-		enc, err := json.Marshal(resp)
-		assert.NoError(t, err)
+		require.Nil(t, sut.km.Start(ctx))
+		<-time.After(2 * time.Second)
 
-		w.WriteHeader(code)
-		_, err = w.Write(enc)
-		assert.NoError(t, err)
+		// key signing is only called once (at start)
+		assert.Equal(t, 1, sut.pdc.CalledCount())
+	})
 
-	}))
-	t.Cleanup(ts.Close)
+	t.Run("new cert requested whenever cert is within expiry window", func(t *testing.T) {
+		ctx := context.Background()
+		// given a keymanager with a cert that is always expired
+		sut := testKeyManager(t)
+		sut.sshCfg.CertCheckCertExpiryPeriod = 100 * time.Millisecond
+		require.Nil(t, sut.km.Start(ctx))
 
-	u, _ = url.Parse(ts.URL)
-	return u, called
+		// leave enough time for the ticker to tick 9 times
+		<-time.After(910 * time.Millisecond)
+
+		// key signing is called a total of 10 times (once at start)
+		assert.Equal(t, 10, sut.pdc.CalledCount())
+
+	})
+}
+
+type mockPDC struct {
+	method string
+	path   string
+	code   int
+	ts     *httptest.Server
+	t      *testing.T
+
+	mu          sync.Mutex
+	calledCount int
```

Comment:
I refactored mockPDC into a struct so we could track the calledCount in a less weird way.

This still assumes that the API only has one path, but we can refactor that if and when we add more.